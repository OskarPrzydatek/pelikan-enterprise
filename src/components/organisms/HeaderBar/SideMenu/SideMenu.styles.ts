import styled, { css } from 'styled-components';

export const SideMenu = styled.div`
  height: 100vh;
  width: 100%;
  background-color: ${({ theme }) => theme.colors.white};
  display: grid;
  grid-template-rows: 1fr 10fr 0.5fr;
  padding: ${({ theme }) => theme.spaces.s};

  @media ${({ theme }) => theme.breakpoints.phone} {
    width: 55%;
  }

  @media ${({ theme }) => theme.breakpoints.tablet} {
    width: 30%;
  }

  @media ${({ theme }) => theme.breakpoints.desktop} {
    width: 20%;
  }
`;

export const HeaderBar = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const Nav = styled.nav``;

export const List = styled.ul``;

export const NavItem = styled.li`
  display: flex;
  flex-direction: column;

  button {
    margin-left: ${({ theme }) => theme.spaces.xl};
    font-weight: 700;
    cursor: pointer;
  }
`;

export const appNameLabelCSS = css`
  color: ${({ theme }) => theme.colors.darkBlue};
  font-weight: 900;
`;

export const currentPageButtonCSS = css`
  color: ${({ theme }) => theme.colors.lightBlue};
`;

export const pageButtonCSS = css`
  color: ${({ theme }) => theme.colors.darkBlue};
`;

export const logoutButtonCSS = css`
  color: ${({ theme }) => theme.colors.white};
`;
