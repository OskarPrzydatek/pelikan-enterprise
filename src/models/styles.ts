import {
  DefaultTheme,
  FlattenInterpolation,
  ThemeProps,
} from 'styled-components';

export interface ICSS {
  css?: FlattenInterpolation<ThemeProps<DefaultTheme>>;
}
