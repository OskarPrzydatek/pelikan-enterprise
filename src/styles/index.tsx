export * from './layouts';
export { theme } from './theme';
