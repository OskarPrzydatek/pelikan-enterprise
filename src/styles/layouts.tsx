import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyleLayout = createGlobalStyle`
  * {
		box-sizing: border-box;
    outline: none;
	} 

  body {
    margin: 0;
    padding: 0;
  }

  #root {
    height: 100vh;
    display: grid;
    grid-template-rows: .5fr 8fr .5fr;
  }

	a {
		color: inherit;
		text-decoration: none;
	}

	ul {
		padding: 0;
    margin: 0;
		list-style-type: none;
	}
`;

export const MainLayout = styled.main`
  padding: ${({ theme }) => theme.spaces.l};
`;

export const PageLayout = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
