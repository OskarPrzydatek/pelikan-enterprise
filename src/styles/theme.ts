import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  colors: {
    red: '#F23557',
    lightBlue: '#98BFF2',
    darkBlue: '#013059',
    gray: '#5F5F5F',
    black: '#2F2F2F',
    opacityBlack: '#2F2F2FCC',
    shadowBlack: '#00000040',
    white: '#FFFFFF',
  },
  breakpoints: {
    phone: '(min-width: 420px)',
    tablet: '(min-width: 920px)',
    desktop: '(min-width: 1200px)',
  },
  fonts: {
    main: `'Inter', sans-serif`,
  },
  fontSizes: {
    xxs: '.2rem',
    xs: '.5rem',
    s: '1rem',
    m: '1.2rem',
    l: '1.5rem',
    xl: '1.8rem',
    xxl: '2rem',
  },
  spaces: {
    xxs: '.2rem',
    xs: '.5rem',
    s: '.8rem',
    m: '1.2rem',
    l: '1.5rem',
    xl: '1.8rem',
    xxl: '2rem',
  },
};
