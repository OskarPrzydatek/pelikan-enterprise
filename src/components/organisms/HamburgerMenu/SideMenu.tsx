import React from 'react';

import { Background, Button, Header, Icon } from '~/components/atoms';
import { Icons, Slugs } from '~/constants';
import { sideMenuNavigationItemsMetadata } from '~/metadata';
import { IComponent, ISideMenuNavigationList } from '~/models';

import * as S from './SideMenu.styles';

interface ISideMenuNavigationItem extends ISideMenuNavigationList {
  handleIsCurrentPage: (path: Slugs) => boolean;
  onNavigate: (path: Slugs) => void;
}

const SideMenuNavigationItem: React.FC<ISideMenuNavigationItem> = ({
  title,
  items,
  handleIsCurrentPage,
  onNavigate,
}: ISideMenuNavigationItem) => {
  const handleIsCurrentPageStyle = (path: Slugs) =>
    handleIsCurrentPage(path) ? S.currentPageButtonCSS : S.pageButtonCSS;

  return (
    <S.NavItem>
      <Header>{title}</Header>
      {items.map(({ path, label }) => (
        <Button
          key={path}
          css={handleIsCurrentPageStyle(path)}
          dataTestID={`side-menu-item-${path}`}
          variant="ghost"
          onClick={() => onNavigate(path)}
        >
          {label}
        </Button>
      ))}
    </S.NavItem>
  );
};

interface ISideMenu extends IComponent {
  onClickCloseSideMenu: () => void;
  handleIsCurrentPage: (path: Slugs) => boolean;
  onNavigate: (path: Slugs) => void;
  onClickLogout: () => void;
}

export const SideMenu: React.FC<ISideMenu> = ({
  onClickCloseSideMenu,
  handleIsCurrentPage,
  onNavigate,
  onClickLogout,
}: ISideMenu) => {
  return (
    <Background onClickBackground={onClickCloseSideMenu}>
      <S.SideMenu data-testid="hamburger-menu-side-menu">
        <S.HeaderBar>
          <Header>Pelikan Enterprise</Header>
          <Icon
            dataTestID="side-menu-close-icon"
            icon={Icons.CLOSE}
            width={15}
            onClick={onClickCloseSideMenu}
          />
        </S.HeaderBar>
        <S.Nav>
          <S.List>
            {sideMenuNavigationItemsMetadata.map(({ title, items }) => (
              <SideMenuNavigationItem
                key={title}
                handleIsCurrentPage={handleIsCurrentPage}
                items={items}
                title={title}
                onNavigate={onNavigate}
              />
            ))}
          </S.List>
        </S.Nav>
        <Button
          css={S.logoutButtonCSS}
          dataTestID="side-menu-logout-button"
          onClick={onClickLogout}
        >
          Wyloguj
        </Button>
      </S.SideMenu>
    </Background>
  );
};
