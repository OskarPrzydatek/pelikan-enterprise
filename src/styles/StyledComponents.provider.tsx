import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from './layouts';
import { theme } from './theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
