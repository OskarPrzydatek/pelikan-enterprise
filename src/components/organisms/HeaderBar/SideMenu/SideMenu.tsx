import React from 'react';

import { Background, Button, Header, Icon, Text } from '~/components/atoms';
import { Icons, Slugs } from '~/constants';
import { IComponent } from '~/models';
import { ILoggedUser } from '~/models/data';

import * as S from './SideMenu.styles';

interface ISideMenuItem {
  slug: Slugs;
  label: string;
  handleIsCurrentPage: (path: Slugs) => boolean;
  onNavigate: (path: Slugs) => void;
}

const SideMenuItem: React.FC<ISideMenuItem> = ({
  slug,
  label,
  handleIsCurrentPage,
  onNavigate,
}: ISideMenuItem) => {
  const handleNavigate = () => onNavigate(slug);
  const handleCurrentPageCSS = (slug: Slugs) =>
    handleIsCurrentPage(slug) ? S.currentPageButtonCSS : S.pageButtonCSS;

  return (
    <S.NavItem>
      <Button
        dataTestID={`side-menu-item-${slug}`}
        variant="ghost"
        onClick={handleNavigate}
      >
        <Header css={handleCurrentPageCSS(slug)}>{label}</Header>
      </Button>
    </S.NavItem>
  );
};

interface ISideMenu extends IComponent {
  onClickCloseSideMenu: () => void;
  handleIsCurrentPage: (path: Slugs) => boolean;
  onNavigate: (path: Slugs) => void;
  onClickLogout: () => void;
}

export const SideMenu: React.FC<ISideMenu> = ({
  onClickCloseSideMenu,
  handleIsCurrentPage,
  onNavigate,
  onClickLogout,
}: ISideMenu) => {
  const loggedUser = JSON.parse(
    localStorage.getItem('loggedUser')!
  ) as ILoggedUser;

  return (
    <Background onClickBackground={onClickCloseSideMenu}>
      <S.SideMenu data-testid="hamburger-menu-side-menu">
        <S.HeaderBar>
          <Text css={S.appNameLabelCSS}>Pelikan Enterprise</Text>
          <Icon
            dataTestID="side-menu-close-icon"
            icon={Icons.CLOSE}
            width={15}
            onClick={onClickCloseSideMenu}
          />
        </S.HeaderBar>
        <S.Nav>
          <S.List>
            <SideMenuItem
              handleIsCurrentPage={handleIsCurrentPage}
              label="Oferty"
              slug={Slugs.OFFER_OVERVIEW}
              onNavigate={onNavigate}
            />
            <SideMenuItem
              handleIsCurrentPage={handleIsCurrentPage}
              label="Hotele"
              slug={Slugs.HOTEL_OVERVIEW}
              onNavigate={onNavigate}
            />
            <SideMenuItem
              handleIsCurrentPage={handleIsCurrentPage}
              label="Transport"
              slug={Slugs.TRANSPORT_OVERVIEW}
              onNavigate={onNavigate}
            />
            <SideMenuItem
              handleIsCurrentPage={handleIsCurrentPage}
              label="Atrakcje"
              slug={Slugs.ATTRACTION_OVERVIEW}
              onNavigate={onNavigate}
            />
            <SideMenuItem
              handleIsCurrentPage={handleIsCurrentPage}
              label="Hashtagi"
              slug={Slugs.HASHTAG_OVERVIEW}
              onNavigate={onNavigate}
            />
            {loggedUser.userType === 'ADMIN' ? (
              <SideMenuItem
                handleIsCurrentPage={handleIsCurrentPage}
                label="UÅ¼ytkownicy"
                slug={Slugs.USERS_OVERVIEW}
                onNavigate={onNavigate}
              />
            ) : null}
          </S.List>
        </S.Nav>
        <Button
          css={S.logoutButtonCSS}
          dataTestID="side-menu-logout-button"
          onClick={onClickLogout}
        >
          Wyloguj
        </Button>
      </S.SideMenu>
    </Background>
  );
};
