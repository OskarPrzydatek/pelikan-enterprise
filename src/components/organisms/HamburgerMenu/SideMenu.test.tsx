import { fireEvent, render, screen } from '@testing-library/react';
import { describe, test, expect, vi } from 'vitest';

import { StyledComponentsProvider } from '~/styles';

import { SideMenu } from './SideMenu.component';

const mockCurrentPageButtonStyle = vi.fn();
const mockOnClickCloseSideMenu = vi.fn();
const mockOnClickLogout = vi.fn();
const mockOnNavigate = vi.fn();

describe('SideMenu', () => {
  test('component snapshot', () => {
    const view = render(
      <StyledComponentsProvider>
        <SideMenu
          currentPageButtonStyle={mockCurrentPageButtonStyle}
          onClickCloseSideMenu={mockOnClickCloseSideMenu}
          onClickLogout={mockOnClickLogout}
          onNavigate={mockOnNavigate}
        />
      </StyledComponentsProvider>
    );
    expect(view).toMatchSnapshot();
  });

  test('ensure navigation works correctly', () => {
    render(
      <StyledComponentsProvider>
        <SideMenu
          currentPageButtonStyle={mockCurrentPageButtonStyle}
          onClickCloseSideMenu={mockOnClickCloseSideMenu}
          onClickLogout={mockOnClickLogout}
          onNavigate={mockOnNavigate}
        />
      </StyledComponentsProvider>
    );
    fireEvent.click(screen.getByTestId('side-menu-item-create-offer'));
    expect(mockOnNavigate).toHaveBeenCalledWith('create-offer');
  });
});
