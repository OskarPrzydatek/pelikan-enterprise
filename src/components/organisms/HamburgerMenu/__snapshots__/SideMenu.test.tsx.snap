// Vitest Snapshot v1

exports[`SideMenu > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div
      class="sc-gswNZR gGMjic"
    >
      <div
        class="sc-ftTHYK cxjxyv"
        data-testid="hamburger-manu-side-menu"
      >
        <div
          class="sc-pyfCe dMgNQh"
        >
          <h1
            class="sc-eDvSVe cDuoXV"
          >
            Pelikan Business
          </h1>
          <button
            class="sc-jSUZER kEKcxS"
            data-testid="side-menu-close-icon"
          >
            <svg
              data-testid="close-icon"
              fill="none"
              viewBox="0 0 20 20"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 2.01429L17.9857 0L10 7.98571L2.01429 0L0 2.01429L7.98571 10L0 17.9857L2.01429 20L10 12.0143L17.9857 20L20 17.9857L12.0143 10L20 2.01429Z"
                fill="#013059"
              />
            </svg>
          </button>
        </div>
        <nav
          class="sc-jrcTuL"
        >
          <ul
            class="sc-kDvujY cWThKJ"
          >
            <li
              class="sc-ipEyDJ jHcbia"
            >
              <h1
                class="sc-eDvSVe cDuoXV"
              >
                Oferty
              </h1>
              <button
                class="sc-dkrFOg hbaBB"
                data-testid="side-menu-item-create-offer"
              >
                Dodaj Ofertę
              </button>
              <button
                class="sc-dkrFOg hbaBB"
                data-testid="side-menu-item-list-offer"
              >
                Lista Ofert
              </button>
            </li>
            <li
              class="sc-ipEyDJ jHcbia"
            >
              <h1
                class="sc-eDvSVe cDuoXV"
              >
                Hotele
              </h1>
              <button
                class="sc-dkrFOg hbaBB"
                data-testid="side-menu-item-create-hotel"
              >
                Dodaj Hotel
              </button>
              <button
                class="sc-dkrFOg hbaBB"
                data-testid="side-menu-item-list-hotel"
              >
                Lista Hoteli
              </button>
            </li>
            <li
              class="sc-ipEyDJ jHcbia"
            >
              <h1
                class="sc-eDvSVe cDuoXV"
              >
                Transport
              </h1>
              <button
                class="sc-dkrFOg hbaBB"
                data-testid="side-menu-item-create-transport"
              >
                Dodaj Transport
              </button>
              <button
                class="sc-dkrFOg hbaBB"
                data-testid="side-menu-item-list-transport"
              >
                Lista Transportów
              </button>
            </li>
            <li
              class="sc-ipEyDJ jHcbia"
            >
              <h1
                class="sc-eDvSVe cDuoXV"
              >
                Atrakcje
              </h1>
              <button
                class="sc-dkrFOg hbaBB"
                data-testid="side-menu-item-create-attraction"
              >
                Dodaj Atrakcję
              </button>
              <button
                class="sc-dkrFOg hbaBB"
                data-testid="side-menu-item-list-attraction"
              >
                Lista Atrakcji
              </button>
            </li>
            <li
              class="sc-ipEyDJ jHcbia"
            >
              <h1
                class="sc-eDvSVe cDuoXV"
              >
                Hash Tagi
              </h1>
              <button
                class="sc-dkrFOg hbaBB"
                data-testid="side-menu-item-create-hashtag"
              >
                Dodaj Hash Tag
              </button>
              <button
                class="sc-dkrFOg hbaBB"
                data-testid="side-menu-item-list-hashtag"
              >
                Lista Hash Tagów
              </button>
            </li>
          </ul>
        </nav>
        <button
          class="sc-dkrFOg GlDzn"
        >
          Wyloguj
        </button>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
